addons/addon-base-raas/packages/base-raas-services/lib/schema/create-study.json

```diff
@@ -38,6 +42,12 @@
       "pattern": "^([A-Fa-f0-9]{40})$",
       "description": "A unique identifier for Open Data in MD5 hash, hexadecimal"
     },
+    "region": {
+      "type": "string",
+      "maxLength": 64,
+      "default": "unknown",
```

Comment:
You could try adding a validation pattern.

Some variations on [https://www.regextester.com/109163](https://www.regextester.com/109163):
- `(us(-gov)?|af|ap|ca|cn|eu|me|sa)-(central|(north|south)?(east|west)?)-\d+[a-z]?` expands allowed country list and adds az letter, matches `us-west-2` or `af-east-1a`.
- `(us(-gov)?|[a-z]{2})-(central|(north|south)?(east|west)?)-\d+[a-z]?` Same as above, but generalizes the country letters for the future

Reply:
I don't think this is a good idea. While validating the data might be attractive, we aren't the authority on what an AWS region will be. Without being able to tie the regex to a published, maintained standard, I think this would be a net negative.

Reply:
TODO: check to see if this gets sanitized

Reply:
TODONE: it does. The DynamoDB rest api handles this.